include/rcpputils/endian.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * If std::endian is not available, the necessary functions are emulated.
+ *
+ * Note: Once std::endian is supported on all ROS2 platforms, this class
+ * can be deprecated in favor of the built-in functionality.
+ *
+ * Note: std::endian is targeted for C++20
+ */
+
+#ifndef RCPPUTILS__ENDIAN_HPP_
+#define RCPPUTILS__ENDIAN_HPP_
+
+#ifdef __has_include
+#  if __has_include(<endian.h>)
+#    include <endian.h>
+#  endif
+#endif
+
+namespace rcpputils
```

Comment:
We may want to wrap this whole thing in another `ifndef` guard, specifically for the C++20 include guard.  Then one day when we switch to C++20, it will just pick that up and use it.  There's an example of doing something like this over at https://github.com/tcbrindle/cpp17_headers/blob/master/include/stx/optional.hpp (also, we may want to add the ability to force-disable it, as that has proven helpful to me in the past).

Reply:
Done in latest.